dissect/target/plugins/os/unix/log/utmp.py

```diff
@@ -68,6 +86,30 @@ def __iter__(self):
 
         while True:
             try:
-                yield utmp.entry(byte_stream)
+                entry = utmp.entry(byte_stream)
+
+                r_type = ""
+                if entry.ut_type in utmp.Type.reverse:
+                    r_type = utmp.Type.reverse[entry.ut_type]
+
+                # UTMP misuses the field ut_addr_v6 for IPv4 and IPv6 addresses, because of this
+                # if the last 12 bytes are zero the value is read as an IPv4-address.
+                if entry.ut_addr_v6[1:] == [0, 0, 0]:
+                    ut_addr = struct.pack("<i", entry.ut_addr_v6[0])
+                else:
+                    ut_addr = struct.pack("<4i", *entry.ut_addr_v6)
```

Comment:
should this logic not be behind a check?
"System V has no ut_host or ut_addr_v6 fields."
https://man7.org/linux/man-pages/man5/utmp.5.html

Reply:
Yes it should

Reply:
I tried to run this plugin on BSD and while the file was parsed without errors the contents did not seem to make sense. Also there seem to be a lot of variations among Linux distros. The only way to determine the file format seems to be to check the version of libc (since that is where utmp.h resides) but that might be quite difficult (i.e. muslc etc..). Another option is to check if the contents contains these 0-bytes because they are a good indicator that the plugin is incompatible, maybe just issue a warning in that case.

Pragmatic approach:
1. Allow the plugin to run on all systems, check for difference SystemV
2. If output records contain 0-bytes (see error handling), emit warning

We can always refine this plug-in later of course, along with all the utmp variations we encounter.

Reply:
Can you create an issue for these utmp/libc variation’s?

Reply:
Can this comment be added to a Github issue outside of this PR?